### PR TITLE
 Improved Bootstrap Process, Documentation, and Repo Automation #32

### LINKED ISSUE #32: Bootstrap missing environment variables for VCS-, TFC authentication
The bootstrap process does not populate Terraform Cloud deployments workspace with environment variables to authenticate to GitHub/Others or Terraform Cloud management. These may have to be abstracted with a `locals` map to support mapping with GITHUB_TOKEN or BITBUCKET_WHATEVER. 

Missing environment variables:
- `GITHUB_TOKEN`: OAuth/PAT to authenticate against GitHub
- `GITHUB_OWNER`: Name of the GitHub Organization to manage
- `TFE_ORGANIZATION`: Name of the Terraform Cloud organization to manage
- `TFE_TOKEN` : ~~Unsure if this must be user, team or organization token~~ **Team tokens work**
- `TF_VAR_vcs_repo_oauth_token_id` | `TF_VAR_vcs_repo_github_app_installation_id`: The oauth token id used by Terraform Cloud to authenticate with repositories for changes (NOT MANAGEMENT).
- `TFE_VAR_tfe_organization_name`: The Terraform Cloud organization to provision resources in (used by Stations station-tfe module)

### PR DESCRIPTION
**Pull request by blinq-kim**
## Overview

This pull request refines the bootstrap process, updates documentation in `readme.md`, and automates GitHub repository creation.

## Details

- **Documentation**: Enhanced `readme.md` with instructions for users to update their values in the bootstrap script.
  
- **Script Enhancements**: 
    - Displays set environment variables (sensitive values hidden).
    - Shows the targeted Azure account.
  
- **Added Environment Variables**: Introduced a few additional environment variables to ensure a comprehensive setup.

- **Creates Github repository**: You will no longer have to manually create a Github repository for the station deployments. This is now a part of the bootstrap process 

- **Confirmation Prompt**: Users are prompted to proceed, providing them an opportunity to review the details.

### CODE REVIEW

**Comment by ghost** [comment]
I have now pushed all my local changes. I think we can add the template repository in another PR later. @blinq-kim

File: bootstrap/README.md
Comment (blinq-kim) [inline bootstrap/README.md:1]: Under "what does it do", make sure GitHub Repository creation is included

File: bootstrap/bootstrap.sh [hunk 1]
```diff
@@ -1,43 +1,78 @@
 #!/bin/bash
```
Comment (blinq-kim) [inline bootstrap/bootstrap.sh:1]: In the future, can we change to tfvars files for configuration and use `jq` to source environment variable values?

File: bootstrap/github_repository.tf [hunk 2]
```diff
@@ -0,0 +1,6 @@
+resource "github_repository" "deployments" {
+  name        = var.vcs_repo_identifier
+  description = "Station deployments"
+
```
Comment (blinq-kim) [inline bootstrap/github_repository.tf]: space?

File: bootstrap/providers.tf
Comment (blinq-kim) [inline bootstrap/providers.tf:1]: Delete this file, it is overwritten in bootstrap.sh anyways.

File: bootstrap/tfe_variables.tf [hunk 3]
```diff
@@ -13,18 +14,62 @@ locals {
...
+      category    = "env"
     },
     ARM_TENANT_ID = {
       value       = data.azurerm_client_config.current.tenant_id
       description = "The Azure Tenant ID to connect to. https://developer.hashicorp.com/terraform/cloud-docs/workspaces/dynamic-provider-credentials/azure-configuration#configure-the-azurerm-or-azuread-provider"
+      sensitive   = false
+      category    = "env"
     },
+    GITHUB_OWNER = {
+      value       = vatfe_organization_namer.vcs_repo_owner
```
Comment (blinq-kim) [inline bootstrap/tfe_variables.tf]: `vatfe_organization_namer.vcs_repo_owner` is invalid code

File: bootstrap/tfe_variables.tf [hunk 4]
```diff
@@ -13,18 +14,62 @@ locals {
...
+    }
+    TFE_TOKEN = {
+      value       = var.tfc_team_token
+      description = "Terraform token for the 'owners' team."
+      sensitive   = true
+      category    = "env"
+    }
+    vcs_repo_github_app_installation_id = {
+      value       = var.vcs_repo_github_app_installation_id
+      description = ""
```
Comment (blinq-kim) [inline bootstrap/tfe_variables.tf]: Missing description

File: bootstrap/tfe_workspace.tf [hunk 5]
```diff
@@ -14,10 +14,10 @@ resource "tfe_workspace" "deployments" {
   description  = "This workspace contains Station Deployments. This workspace was bootstrapped from ${var.bootstrap_repo_url}"
 
   vcs_repo {
-    identifier                 = var.vcs_repo_identifier
-    branch                     = try(var.vcs_repo_branch, null)
-    oauth_token_id             = var.vcs_repo_oauth_token_id
-    github_app_installation_id = var.vcs_repo_github_app_installation_id
+    identifier                 = github_repository.deployments.name
+    branch                     = github_repository.deployments.default_branch
+    oauth_token_id             = try(var.vcs_repo_oauth_token_id, null)
```
Comment (blinq-kim) [inline bootstrap/tfe_workspace.tf]: Default value is null in variables.tf file, try not needed.

File: bootstrap/tfe_workspace.tf [hunk 6]
```diff
@@ -14,10 +14,10 @@ resource "tfe_workspace" "deployments" {
...
+    github_app_installation_id = try(var.vcs_repo_github_app_installation_id, null)
```
Comment (blinq-kim) [inline bootstrap/tfe_workspace.tf]: Default value is null in variables.tf file, try not needed.

File: bootstrap/tfe_workspace.tf [hunk 7]
```diff
@@ -14,10 +14,10 @@ resource "tfe_workspace" "deployments" {
...
     tags_regex                 = try(var.vcs_repo_tags_regex, null)
```
Comment (blinq-kim) [inline bootstrap/tfe_workspace.tf]: Default value is null in variables.tf file, try not needed. (even if this wasn't added in this PR, fix this as well)

File: bootstrap/providers.tf
Comment (ghost) [reply bootstrap/providers.tf:1]: Deleted and added to .gitignore

File: bootstrap/README.md
Comment (blinq-kim) [reply bootstrap/README.md:1]: @Blinq-SanderBlom - this is still missing.

**Pull request merged**